sympy/geometry/ellipse.py

```diff
@@ -327,16 +328,17 @@ def circumference(self):
         >>> p1 = Point(0, 0)
         >>> e1 = Ellipse(p1, 3, 1)
         >>> e1.circumference
-        12*Integral(sqrt((-8*_x**2/9 + 1)/(-_x**2 + 1)), (_x, 0, 1))
+        12*elliptic_e(2*sqrt(2)/3)
 
         """
-        from sympy import Integral
         if self.eccentricity == 1:
-            return 2*pi*self.hradius
+            # degenerate
+            return 4 * self.major
+        elif self.eccentricity == 0:
+            # circle
+            return 2 * pi * self.hradius
         else:
-            x = Dummy('x', real=True)
-            return 4*self.major*Integral(
-                sqrt((1 - (self.eccentricity*x)**2)/(1 - x**2)), (x, 0, 1))
+            return 4 * self.major * elliptic_e(self.eccentricity)
```

Comment:
This turns out to be incorrect.  The correct formula is `4 * e.major * elliptic_e(e.eccentricity**2)` 

There should be some kind of a numeric test for correctness, so we don't depend just on someone's ability to catch such mistakes. [Wolfram Alpha](https://www.wolframalpha.com/input/?i=ellipse+with+a%3D5,+b%3D3) gives numeric values for ellipse perimenter